src/storage.rs

```diff
@@ -1836,11 +1843,11 @@ impl Cache {
 
                 // Check if the resource is already available in our local cache.
                 let fetched_package_ = fetched_package.clone();
+                let now = filetime::FileTime::from_system_time(SystemTime::from(self.now));
                 let cached_file = tokio::task::spawn_blocking(move || {
                     File::open(&fetched_package_).map(|file| {
                         // Update the atime and mtime for this crate to ensure it isn't
                         // collected by the gc.
-                        let now = filetime::FileTime::now();
                         if let Err(err) =
```

Comment:
hmm, this is a bit weird because it does kinda mess with the mtimes of files in the local cache checkout, but that's probably fine as we aren't checking them in anyway? I don't think we run the gc on normal operations yet, but it might become more of a thing if/when we start doing that.

Reply:
This change gave me pause for a bit too, but I think for now it should be okay (and I _think_ it's more preferable than using some other time source?). It's simple enough to revert in the future.